adr/ADR-20.md

```diff
@@ -38,60 +38,98 @@ Possible future features
 - Per chunk Content-Encoding (i.e. gzip)
 - Read an individual chunk. 
 
+## Basic Design
+
+- Object store or bucket is backed by a stream
+- Multiple objects can be placed in each bucket
+- Object Info is stored as json in the payload of the message on the Object Info message subject.
+- The Object Info subject is always rolled up (per subject)
+- Object chunks are stored as the payload of messages on the Chunk message subject
+
 ## Naming Specification
 
 Protocol Naming Conventions are fully defined in [ADR-6](ADR-6.md)
 
 ### Object Store
-The object store name or bucket name (`os-bucket-name`) will be used to formulate a stream name and is specified as: `restricted-term` or 1 or more of `A-Z, a-z, 0-9, dash, underscore`
+The object store name or bucket name (`bucket`) will be used to formulate a stream name 
+and is specified as: `restricted-term` (1 or more of `A-Z, a-z, 0-9, dash, underscore`)
 
-### Objects
-An individual object name is not restricted. It is base64 encoded to form the (`os-object-name`).
+### Object Id
+Object ids (`object-nuid`) are a nuid.
 
-### Chunk Ids
-Chunk ids (`chunk-id`) should be a nuid.
+### Object Name
+An individual object name is not restricted. It is base64 encoded to form `name-encoded`.
 
-### Component Templates
+### Digest
+Currently `SHA-256` is the only supported digest. Please use the uppercase form as in [RFC-6234](https://www.rfc-editor.org/rfc/rfc6234)
+when specifying the digest as in `SHA-256=3F9239B0272558CE6E3E98731C43A654AC6882C5050D5F352B2A5BFBB8DE0058`.
```

Comment:
Note the hash may have padding `=` or `==` or `===`

Reply:
fixed example